js/ag_reference.js

```diff
@@ -905,3 +921,181 @@ AttributeGroupReferenceAggregateFn.prototype = {
+};
+
+BucketAttributeGroupReference.prototype.search = function (term) {
+    verify(true == false, "Invalid function");
+};
+
+/**
+ * @return {ERMRest.BucketAttributeGroupPage}
+ */
+BucketAttributeGroupReference.prototype.read = function () {
+    var moment = module._moment;
+
+    // function decimals_s(num){
+    //     var num_s = "" + num;
+    //     return num_s.split(".")[1].length;
+    // }
+    //
+    // function addFloats(min, width) {
+    //     var precision = Math.max(decimals_s(min), decimals_s(width));
+    //     return ((min*1) + (width*1)).toFixed(precision);
+    // }
+
+    function calculateWidthLabel(min, binWidth) {
+        var nextLabel;
+        if (currRef._keyColumns[0].type.rootName.indexOf("date") > -1)  {
+            nextLabel = moment(min).add(binWidth, 'd').format(module._dataFormats.DATE);
+        } else if (currRef._keyColumns[0].type.rootName.indexOf("timestamp") > -1) {
+            nextLabel = moment(min).add(binWidth, 's').format(module._dataFormats.DATETIME.display);
+        } else {
+            nextLabel = (min + binWidth);
+        }
+        return nextLabel;
+    }
+
+    try {
+        var defer = module._q.defer();
+
+        var uri = this.uri;
+
+        var currRef = this;
+        this._server._http.get(uri).then(function (response) {
+            var data = {
+                x: [],
+                y: []
+            };
+
+            var labels = {
+                min: [],
+                max: []
+            };
+
+            var min, max;
+
+            for (var i=0; i<response.data.length; i++) {
+                var index = response.data[i].c1[0];
+                if (index !== null) {
+                    min = response.data[i].c1[1];
+                    max = response.data[i].c1[2];
+
+                    if (currRef._keyColumns[0].type.rootName.indexOf("date") > -1) {
+                        min = min !== null ? moment(min).format(module._dataFormats.DATE) : null;
+                        max = max !== null ? moment(max).format(module._dataFormats.DATE) : null;
+                    } else if (currRef._keyColumns[0].type.rootName.indexOf("timestamp") > -1) {
+                        min = min !== null ? moment(min).format(module._dataFormats.DATETIME.display) : null;
+                        max = max !== null ? moment(max).format(module._dataFormats.DATETIME.display) : null;
+                    }
+
+                    labels.min[index] = min;
+                    labels.max[index] = max;
+
+                    data.x[index] = min;
+                    data.y[index] = response.data[i].c2;
+                }
+                // else if null (this is the null bin)
+                // we currently don't want to do anything with the null values
+            }
+
+            // This should be set to 12 to include the # of bins we want to display + the above max and below min bucket
+            for (var j=0; j<currRef._options.bucketCount+2; j++) {
```

Comment:
If bucketCount and the other two options are actually required and not optional just flat that list out. Instead of having them in a options object, list them as the parameters of the constructor.

Reply:
They aren't required or used by `chaise` in any way. Those properties are required for calculating the labels to return to chaise though and padding out the x and y axes.

I figured it would be easier to add one object (maybe `_options` is the wrong name) that holds the values I need for calculations so the constructor only includes one value instead of 4 seemingly extra ones.